File: hotelaria-frontend/src/app/shared/enums/TipoTransacaoEnum.ts
export class TipoTransacaoEnum {
    static ENTRADA = new TipoTransacaoEnum("Entrada");
    static SAIDA = new TipoTransacaoEnum("Saída");
    static BAIXA = new TipoTransacaoEnum("Baixa");
    static ESTORNO = new TipoTransacaoEnum("Estorno");

    private constructor(private descricao: string) {}

    getDescricao(): string {
        return this.descricao;
    }
}

File: Hotelaria/src/main/java/com/dolts/hotelaria/enums/TipoTransacaoEnum.java
package com.dolts.hotelaria.enums;

import lombok.Getter;

@Getter
public enum TipoTransacaoEnum {
    ENTRADA("Entrada"),
    SAIDA("Saída"),
    BAIXA("Baixa"),
    ESTORNO("Estorno");

    private final String descricao;

    TipoTransacaoEnum(String descricao) {
        this.descricao = descricao;
    }

}
